hooks/pre-command

```diff
@@ -37,6 +39,14 @@ upsert_ecr() {
   aws ecr put-lifecycle-policy \
   --repository-name "${repository_name}" \
   --lifecycle-policy-text "file://${lifecycle_policy_file}"
+
+  if [[ "${BUILDKITE_PLUGIN_CREATE_ECR_SCAN_ON_PUSH:-}" =~ ^(true|on|1)$ ]]; then
```

Comment:
Ah sorry, I misled you. We'll need to accept falsy values too, and then map then back to one of `true` or `false` for the AWS CLI command to work.

If it keeps the code simpler, we could do this with an if else:

```shell
if [[ "${BUILDKITE_PLUGIN_CREATE_ECR_SCAN_ON_PUSH:-}" =~ ^(true|on|1)$ ]]; then
  aws ecr ... scanOnPush=true
elif [[ "${BUILDKITE_PLUGIN_CREATE_ECR_SCAN_ON_PUSH:-}" =~ ^(false|off|0)$ ]]; then
  aws ecr ... scanOnPush=false
fi
```

Reply:
Agree. I have covered both enabling and disabling of image scan but slightly different way just to reduce the lines repetition.